### PR TITLE
Feature/geode 4087

### PR DESCRIPTION
**Pull request by bschuchardt**
*GEODE-4087 modify SocketCreator to not set the default SSLContext for the JVM*

_SocketCreator no longer installs a default SSL context._

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.
@jinmeiliao @PurelyApplied @PivotalSarge @galen-pivotal @upthewaterspout @WireBaron

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/JettyHelper.java [hunk 1]
```diff
@@ -32,8 +33,14 @@
 import org.eclipse.jetty.webapp.WebAppContext;
 
 import org.apache.geode.GemFireConfigException;
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.distributed.internal.DistributionConfigImpl;
 import org.apache.geode.internal.admin.SSLConfig;
 import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.internal.net.SSLConfigurationFactory;
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/JettyHelper.java]: Remove unused imports.

File: geode-core/src/test/java/org/apache/geode/management/internal/JettyHelperJUnitTest.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import org.apache.geode.GemFireConfigException;
 import org.apache.geode.distributed.internal.DistributionConfigImpl;
 import org.apache.geode.internal.net.SSLConfigurationFactory;
+import org.apache.geode.internal.net.SocketCreatorFactory;
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/internal/JettyHelperJUnitTest.java]: Unused import.

File: geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java [hunk 3]
```diff
@@ -40,12 +40,14 @@
...
 import javax.management.remote.rmi.RMIJRMPServerImpl;
 import javax.management.remote.rmi.RMIServerImpl;
+import javax.net.ssl.SSLContext;
 import javax.rmi.ssl.SslRMIClientSocketFactory;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.logging.log4j.Logger;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java]: Unused imports

File: geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java [hunk 4]
```diff
@@ -235,8 +237,10 @@ private void startHttpService(boolean isServer) {
 
           boolean isRestWebAppAdded = false;
 
-          this.httpServer = JettyHelper.initJetty(bindAddress, port,
-              SSLConfigurationFactory.getSSLConfigForComponent(SecurableCommunicationChannel.WEB));
+          SocketCreator webSocketCreator =
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java]: webSocketCreator appears to be unused.

File: geode-assembly/src/test/java/org/apache/geode/tools/pulse/PulseSecurityWithSSLTest.java [hunk 5]
```diff
@@ -35,6 +35,9 @@
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
+import org.apache.geode.distributed.internal.DistributionConfigImpl;
+import org.apache.geode.internal.net.SocketCreatorFactory;
+import org.apache.geode.internal.security.SecurableCommunicationChannel;
```
Comment (pdxrunner) [inline geode-assembly/src/test/java/org/apache/geode/tools/pulse/PulseSecurityWithSSLTest.java]: Unused import

File: geode-core/src/main/java/org/apache/geode/management/internal/RestAgent.java [hunk 6]
```diff
@@ -33,6 +33,7 @@
 import org.apache.geode.internal.logging.LogService;
 import org.apache.geode.internal.net.SSLConfigurationFactory;
 import org.apache.geode.internal.net.SocketCreator;
+import org.apache.geode.internal.net.SocketCreatorFactory;
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/RestAgent.java]: Unused imports.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java [hunk 7]
```diff
@@ -21,8 +21,11 @@
 import org.awaitility.Awaitility;
 import org.junit.rules.ExternalResource;
 
+import org.apache.geode.distributed.internal.DistributionConfigImpl;
 import org.apache.geode.internal.AvailablePort;
 import org.apache.geode.internal.admin.SSLConfig;
```
Comment (pdxrunner) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: This is now an unused import

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java [hunk 8]
```diff
@@ -350,7 +350,16 @@ private void initialize() {
...
+          // if (SecurableCommunicationChannel.JMX
+          // .equals(sslConfig.getSecuredCommunicationChannel())) {
+          // // JMX connections use the default SSL context so we must override it
+          // if (System.getProperty("javax.net.ssl.keyStore") != null) {
+          // logger.warn("Overriding the default SSL context that was established through system
+          // properties");
+          // }
+          // logger.error("Setting a default SSLContext", new Exception());
+          // SSLContext.setDefault(sslContext);
+          // }
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: I would remove the commented out code.

**Comment by bschuchardt** [comment]
I'm closing this PR and deleting the current feature/GEODE-4087 branch.  The changes I made to Jetty startup caused problems with Pulse tests and aren't really necessary.  I'll create a new branch with a slimmer fix and a test.

**Pull request closed without merging**